shuttle/tests/basic/once.rs

```diff
@@ -249,3 +249,25 @@ fn poison() {
         None,
     );
 }
+
+// Tests that `Once` state is preserved even if the `Once` is moved.
+fn moved<F>(checker: F)
+where
+    F: FnOnce(Box<dyn Fn() + Send + Sync + 'static>),
+{
+    checker(Box::new(move || {
+        let once = Box::new(Once::new());
+        assert!(!once.is_completed());
+        once.call_once(|| {});
+        assert!(once.is_completed());
+
+        // The `Once` is still completed, even if its moved to another address.
+        let once = Box::new(*once);
+        assert!(once.is_completed());
```

Comment:
Might be pedantic, but should we create a second `Once` and assert that it is not completed (ie that `completed` does not apply to all `Once`s?)